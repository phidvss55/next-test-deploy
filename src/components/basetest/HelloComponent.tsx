import React from 'react'

interface Props {
  name?: string | undefined
}

const HelloComponent = ({ name }: Props) => {
  return (
    <div>
      <h1>Hello Component {name}</h1>
    </div>
  )
}

export default HelloComponent
