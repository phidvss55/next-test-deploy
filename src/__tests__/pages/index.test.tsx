import HelloComponent from "@/components/test/HelloComponent"
import { render, screen } from '@testing-library/react'

describe('Index Page', () => {
  it('Should render property', () => {
    render(<HelloComponent name={'User'} />)
    const header = screen.getByRole('heading')
    expect(header).toHaveTextContent('Hello')
  })

  it('Should have a button', () => {
    render(<HelloComponent />)
    const button = screen.queryAllByRole('button', { hidden: true })
    expect(button.length).toBeLessThanOrEqual(1)
  })

  it('renders homepage unchanged', () => {
    const { container } = render(<HelloComponent />);
    expect(container).toMatchSnapshot();
  });
})