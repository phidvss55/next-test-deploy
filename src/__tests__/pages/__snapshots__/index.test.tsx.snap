// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Page renders homepage unchanged 1`] = `
<div>
  <div>
    <h1>
      Hello Component
    </h1>
  </div>
</div>
`;
